plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergStatistics.java

```diff
@@ -0,0 +1,328 @@
+
+    private IcebergStatistics(
+            long recordCount,
+            long fileCount,
+            long size,
+            Map<Integer, Object> minValues,
+            Map<Integer, Object> maxValues,
+            Map<Integer, Long> nullCounts,
+            Map<Integer, Long> columnSizes)
+    {
+        this.recordCount = recordCount;
+        this.fileCount = fileCount;
+        this.size = size;
+        this.minValues = requireNonNull(minValues, "minValues is null");
+        this.maxValues = requireNonNull(maxValues, "maxValues is null");
+        this.nullCounts = requireNonNull(nullCounts, "nullCounts is null");
+        this.columnSizes = requireNonNull(columnSizes, "columnSizes is null");
+    }
+
+    public long getRecordCount()
+    {
+        return recordCount;
+    }
+
+    public long getFileCount()
+    {
+        return fileCount;
+    }
+
+    public long getSize()
+    {
+        return size;
+    }
+
+    public Map<Integer, Object> getMinValues()
+    {
+        return minValues;
+    }
+
+    public Map<Integer, Object> getMaxValues()
+    {
+        return maxValues;
+    }
+
+    public Map<Integer, Long> getNullCounts()
+    {
+        return nullCounts;
+    }
+
+    public Map<Integer, Long> getColumnSizes()
+    {
+        return columnSizes;
+    }
+
+    public static class IcebergStatisticsBuilder
+    {
+        private final Map<Integer, Type.PrimitiveType> idToTypeMapping;
+        private final List<Types.NestedField> columns;
+        private final TypeManager typeManager;
+        private final Map<Integer, Optional<Long>> nullCounts = new HashMap<>();
+        private final Map<Integer, ColumnStatistics> columnStatistics = new HashMap<>();
+        private final Map<Integer, Long> columnSizes = new HashMap<>();
+        private long recordCount;
+        private long fileCount;
+        private long size;
+
+        public IcebergStatisticsBuilder(
+                Map<Integer, Type.PrimitiveType> idToTypeMapping,
+                List<Types.NestedField> columns,
+                TypeManager typeManager)
+        {
+            this.idToTypeMapping = requireNonNull(idToTypeMapping, "idToTypeMapping is null");
+            this.columns = requireNonNull(columns, "columns is null");
+            this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        }
+
+        public void registerDataFile(DataFile dataFile, PartitionSpec partitionSpec)
+        {
+            fileCount++;
+            recordCount += dataFile.recordCount();
+            size += dataFile.fileSizeInBytes();
+
+            Map<Integer, Long> newColumnSizes = dataFile.columnSizes();
+            if (newColumnSizes != null) {
+                for (Types.NestedField column : columns) {
+                    int id = column.fieldId();
+                    Long addedSize = newColumnSizes.get(id);
+                    if (addedSize != null) {
+                        columnSizes.merge(id, addedSize, Long::sum);
```

Comment:
Does Iceberg report column size for all types, or only variable width?

we could limit the calculations to variable width only, as otherwise the stat is ignored by the engine.

(optional, really_